pkg/splunk/enterprise/searchheadcluster.go

```diff
@@ -59,6 +59,13 @@ func ApplySearchHeadCluster(client splcommon.ControllerClient, cr *enterprisev1.
 	}
 	if cr.Status.ShcSecretChanged == nil {
 		cr.Status.ShcSecretChanged = []bool{}
+
+	}
+	if cr.Status.AdminSecretChanged == nil {
+		cr.Status.AdminSecretChanged = []bool{}
+	}
+	if cr.Status.AdminPasswordChangedSecrets == nil {
+		cr.Status.AdminPasswordChangedSecrets = make(map[string]bool)
 	}
```

Comment:
Nit: Now, we have multiple inits, probably, we need to have a separate init function. But it's not a blocker at this time.